logdetective/server/models.py

```diff
@@ -37,40 +37,59 @@ class JobHook(BaseModel):
     project_id: int
 
 
+class Explanation(BaseModel):
+    """Model of snippet or general log explanation from Log Detective"""
+
+    text: str
+    logprobs: Optional[List[Dict]] = None
+
+    def __str__(self):
+        return self.text
+
+
+class AnalyzedSnippet(BaseModel):
+    """Model for snippets already processed by Log Detective.
+
+    explanation: LLM output in form of plain text and logprobs dictionary
+    text: original snippet text
+    line_number: location of snippet in original log
+    """
+    explanation: Explanation
+    text: str
+    line_number: int
+
+
 class Response(BaseModel):
     """Model of data returned by Log Detective API
 
-    explanation: CreateCompletionResponse
-        https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.llama_types.CreateCompletionResponse
+    explanation: Explanation
     response_certainty: float
     """
 
-    explanation: Dict
+    explanation: Explanation
     response_certainty: float
 
 
 class StagedResponse(Response):
     """Model of data returned by Log Detective API when called when staged response
     is requested. Contains list of reponses to prompts for individual snippets.
 
-    explanation: CreateCompletionResponse
-        https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.llama_types.CreateCompletionResponse
+    explanation: Explanation
     response_certainty: float
-    snippets:
-        list of dictionaries {
-        'snippet' : '<original_text>,
-        'comment': CreateCompletionResponse,
-        'line_number': '<location_in_log>' }
+    snippets: list of AnalyzedSnippet objects
     """
 
-    snippets: List[Dict[str, str | Dict | int]]
+    snippets: List[AnalyzedSnippet]
```

Comment:
This is lovely that it got its own model

Reply:
Just so we're clear, this is coming from #139 
I rebased this set of patches atop those because it vastly simplifies what I needed to do here.